// obs-websocket-js/dist/msgpack.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true}); var _class;





var _chunkD6NG7GDDcjs = require('./chunk-D6NG7GDD.cjs');

// src/msgpack.ts
var _msgpack = require('@msgpack/msgpack');
var OBSWebSocket = (_class = class extends _chunkD6NG7GDDcjs.BaseOBSWebSocket {constructor(...args) { super(...args); _class.prototype.__init.call(this); }
  __init() {this.protocol = "obswebsocket.msgpack"}
  async encodeMessage(data) {
    return _msgpack.encode.call(void 0, data);
  }
  async decodeMessage(data) {
    if (typeof Blob !== "undefined" && data instanceof Blob) {
      data = await data.arrayBuffer();
    }
    return _msgpack.decode.call(void 0, data);
  }
}, _class);
var msgpack_default = OBSWebSocket;







exports.EventSubscription = _chunkD6NG7GDDcjs.EventSubscription; exports.OBSWebSocket = OBSWebSocket; exports.OBSWebSocketError = _chunkD6NG7GDDcjs.OBSWebSocketError; exports.RequestBatchExecutionType = _chunkD6NG7GDDcjs.RequestBatchExecutionType; exports.WebSocketOpCode = _chunkD6NG7GDDcjs.WebSocketOpCode; exports.default = msgpack_default;
